impl/src/main/java/io/kroxylicious/testing/kafka/common/Utils.java

```diff
@@ -163,6 +168,22 @@ public static void awaitExpectedBrokerCountInClusterViaTopic(Map<String, Object>
         }
     }
 
+    public static void createUsersOnClusterIfNecessary(Map<String, Object> connectionConfig, KafkaClusterConfig clusterConfig) {
+        var users = clusterConfig.getUsers();
+        var saslMechanism = clusterConfig.getSaslMechanism();
+        if (users.isEmpty() || !clusterConfig.isSaslScram()) {
+            return;
+        }
+        var sci = new ScramCredentialInfo(ScramMechanism.fromMechanismName(saslMechanism), SCRAM_ITERATIONS);
+        try (var admin = Admin.create(connectionConfig)) {
+            // TODO fail gracefully if KRaft and Metaddata version does not yet support SCRAM
```

Comment:
This is https://issues.apache.org/jira/browse/KAFKA-14084 - if under KRaft we need at least Kafka 3.5.0.  If no we should just fail.

Reply:
Actually, we haven't got a clean way to detect whether SCRAM will be supported.  For KRaft we need to know the storage format used by the Broker which we don't have a programmatic way to tell.   tbh I don't think this matters too much.  I think we just document in the release notes.

- container - must use a new kafka-native image
- in-vm - use a kafka version >= 3.5.0

I think that's good enough.